client/sites/staging-site/components/staging-site-sync-modal/index.tsx

```diff
@@ -57,23 +53,28 @@ const DirectionArrow = () => {
 interface EnvironmentLabelProps {
 	label: string;
 	environmentType: EnvironmentType;
+	siteTitle?: string;
 }
 
-const EnvironmentLabel = ( { label, environmentType }: EnvironmentLabelProps ) => {
+const EnvironmentLabel = ( { label, environmentType, siteTitle }: EnvironmentLabelProps ) => {
 	return (
-		<VStack spacing={ 1 } style={ { flex: 1 } }>
+		<VStack spacing={ 1 }>
 			<SectionHeader level={ 3 } title={ label } />
-			<InputControl
-				readOnly
-				prefix={
-					<InputControlPrefixWrapper>
-						<SiteEnvironmentBadge environmentType={ environmentType } />
-					</InputControlPrefixWrapper>
-				}
-				__next40pxDefaultSize
-				tabIndex={ -1 }
-				aria-hidden="true"
-			/>
+			<HStack spacing={ 2 }>
+				<SiteEnvironmentBadge environmentType={ environmentType } />
+				{ siteTitle && (
+					<Text
+						style={ {
+							whiteSpace: 'nowrap',
+							overflow: 'hidden',
+							textOverflow: 'ellipsis',
+							maxWidth: '190px',
+						} }
+					>
+						{ siteTitle }
+					</Text>
```

Comment:
I wanted to use Text component props (https://wordpress.github.io/gutenberg/?path=/docs/components-experimental-text--docs) meant for ellipsis, but they do not seem to work.